src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -1850,33 +1873,51 @@ internal static void AddDiscriminatorMethod(CodeClass newClass, string discrimin
+                                    var shortestNamespaceName = GetModelsNamespaceNameFromReferenceId(propertySchema.Reference?.Id);
+                                    var targetNamespace = string.IsNullOrEmpty(shortestNamespaceName) ? ns :
+                                                        rootNamespace?.FindOrAddNamespace(shortestNamespaceName) ?? ns;
+                                    var definition = CreateModelDeclarations(currentNode, propertySchema, default, targetNamespace, string.Empty, typeNameForInlineSchema: className);
+                                    if (definition == null)
                                     {
-                                        var propertySchema = x.Value;
-                                        var className = propertySchema.GetSchemaName().CleanupSymbolName();
-                                        if (string.IsNullOrEmpty(className))
-                                            className = $"{model.Name}_{x.Key.CleanupSymbolName()}";
-                                        var shortestNamespaceName = GetModelsNamespaceNameFromReferenceId(propertySchema.Reference?.Id);
-                                        var targetNamespace = string.IsNullOrEmpty(shortestNamespaceName) ? ns :
-                                                            rootNamespace?.FindOrAddNamespace(shortestNamespaceName) ?? ns;
-                                        var definition = CreateModelDeclarations(currentNode, propertySchema, default, targetNamespace, string.Empty, typeNameForInlineSchema: className);
-                                        if (definition == null)
-                                        {
-                                            logger.LogWarning("Omitted property {PropertyName} for model {ModelName} in API path {ApiPath}, the schema is invalid.", x.Key, model.Name, currentNode.Path);
-                                            return null;
-                                        }
-                                        return CreateProperty(x.Key, definition.Name, propertySchema: propertySchema, existingType: definition);
-                                    })
-                                    .OfType<CodeProperty>()
-                                    .ToArray());
+                                        logger.LogWarning("Omitted property {PropertyName} for model {ModelName} in API path {ApiPath}, the schema is invalid.", x.Key, model.Name, currentNode.Path);
+                                        return null;
+                                    }
+                                    return CreateProperty(x.Key, definition.Name, propertySchema: propertySchema, existingType: definition);
+                                })
+                                .OfType<CodeProperty>()
+                                .ToArray());
+        }
+    }
+    private Dictionary<String, OpenApiSchema> CollectAllProperties(OpenApiSchema schema)
+    {
```

Comment:
I'm not sure I understand what's the goal here?
Why do you need to get all the properties of the allOf entries? Aren't you going to grab properties from the parent type? if so shouldn't you do it recursively?

Reply:
No, this is not to get the properties from the parent types (that's why it's not collecting properties from referenced schema's). It's to collect the properties defined in the type itself. A root-type will have the properties directly in its schema, but a type that extends another type has an `allOf` with the properties. This `allOf` will typically only have 2 parts: a reference to the super type and the contents of the current type, but that second part could potentially be split in more parts.

I had to rewrite this part of the code anyway to get the correct call to `model.AddPropety`, so I decided to make it a bit more resilient against strange `allOf` constructions. This should probably fix this test: https://github.com/microsoft/kiota/pull/2769/files#diff-ca13d201dbddd478e6799e4ae772c20dca0553957883e8440cc585727d28b387R6212